test/extensions/filters/http/decompressor/decompressor_filter_test.cc

```diff
@@ -111,70 +145,73 @@ INSTANTIATE_TEST_SUITE_P(IsRequestDirection, DecompressorFilterTest,
                          ::testing::Values(true, false));
 
 TEST_P(DecompressorFilterTest, DecompressionActive) {
-  // Keep the decompressor to set expectations about it
-  auto decompressor = std::make_unique<Compression::Decompressor::MockDecompressor>();
-  auto* decompressor_ptr = decompressor.get();
-  EXPECT_CALL(*decompressor_factory_, createDecompressor())
-      .WillOnce(Return(ByMove(std::move(decompressor))));
   Http::TestHeaderMapImpl headers_before_filter{{"content-encoding", "mock"},
                                                 {"content-length", "256"}};
-  std::unique_ptr<Http::RequestOrResponseHeaderMap> headers_after_filter =
-      doHeaders(headers_before_filter, false /* end_stream */);
-
-  EXPECT_EQ(nullptr, headers_after_filter->ContentEncoding());
+  decompressionActive(headers_before_filter, absl::nullopt /* expected_content_encoding */);
+}
 
-  // FIX ME: pending decision on this.
-  EXPECT_EQ(nullptr, headers_after_filter->ContentLength());
-  EXPECT_EQ("chunked", headers_after_filter->TransferEncoding()->value().getStringView());
-  const auto* accept_encoding = headers_after_filter->get(Http::LowerCaseString{"accept-encoding"});
-  if (isRequestDirection()) {
-    EXPECT_EQ("mock", accept_encoding->value().getStringView());
-  } else {
-    EXPECT_EQ(nullptr, accept_encoding);
-  }
+TEST_P(DecompressorFilterTest, DecompressionActiveContentEncodingSpacing) {
+  // Additional spacing should still match.
+  Http::TestHeaderMapImpl headers_before_filter{{"content-encoding", " mock "},
+                                                {"content-length", "256"}};
+  decompressionActive(headers_before_filter, absl::nullopt /* expected_content_encoding */);
+}
 
-  expectDecompression(decompressor_ptr);
+TEST_P(DecompressorFilterTest, DecompressionActiveContentEncodingCasing) {
+  // Different casing should still match.
+  Http::TestHeaderMapImpl headers_before_filter{{"content-encoding", "MOCK"},
```

Comment:
Are keys forced to be lowercased at the point they're passed to filters? If not, we should probably handle that as well (though I assume they are sanitized correctly)

Reply:
yep, this is all avoided because keys to headers are actually instances of the LowerCaseString class. This is just a convenience constructor.